Mathlib/RingTheory/PowerSeries/Order.lean

```diff
@@ -303,30 +311,58 @@ theorem order_X_pow (n : ℕ) : order ((X : R⟦X⟧) ^ n) = n := by
   rw [X_pow_eq, order_monomial_of_ne_zero]
   exact one_ne_zero
 
+@[simp]
+theorem divided_by_X_pow_order_of_X_eq_one :
+    divided_by_X_pow_order X_ne_zero = (1 : R⟦X⟧) := by
+  apply X_pow_mul_cancel
+  rw [self_eq_X_pow_order_mul_divided_by_X_pow_order]
+  simp
+
 end OrderZeroNeOne
 
-section OrderIsDomain
+section NoZeroDivisors
 
--- TODO: generalize to `[Semiring R] [NoZeroDivisors R]`
-variable [CommRing R] [IsDomain R]
+variable [Semiring R] [NoZeroDivisors R]
 
 /-- The order of the product of two formal power series over an integral domain
```

Comment:
Do you want to change "integral domain"? I guess that's the most common case still, so I won't insist

Reply:
I just wanted to address the `TODO`.
And `IsDomain` adds `Nontrivial` which is not always needed.